raft/log.go

```diff
@@ -171,6 +177,27 @@ func (l *Log) State() State {
 	return l.state
 }
 
+// Index returns the highest available index.
+func (l *Log) Index() uint64 {
+	l.mu.Lock()
```

Comment:
We're using a `sync.Mutex`. We don't have a lot of concurrency in the log so I don't think an `RWMutex` would help us and it takes longer to do uncontended locks with an `RWMutex` than it does with a `Mutex`.